core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -31,19 +33,21 @@ import org.apache.kafka.common.utils.Time
 
 import scala.collection.JavaConverters._
 import scala.collection._
+import scala.collection.mutable.ArrayBuffer
 
 /**
  * The entry point to the kafka log management subsystem. The log manager is responsible for log creation, retrieval, and cleaning.
  * All read and write operations are delegated to the individual log instances.
- * 
+ *
  * The log manager maintains logs in one or more directories. New logs are created in the data directory
  * with the fewest logs. No attempt is made to move partitions after the fact or balance based on
  * size or I/O rate.
- * 
+ *
  * A background thread handles log retention by periodically truncating excess log segments.
  */
 @threadsafe
-class LogManager(val logDirs: Array[File],
+class LogManager(private val logDirs: Array[File],
+                 private val initialOfflineDirs: Array[File],
```

Comment:
I think it is probably not necessary to call this on initialOfflineDirs. The purpose of calling `maybeAddLogFailureEvent()` is to cleanup state (e.g. LogManager.logs) and notifying controller. For initialOfflineDirs, the no state will be created for replicas in these offline log directories. And the broker hasn't registered itself in the zookeeper yet and thus controller will query this broker for state of all replicas afterwards -- thus no need to notify controller either.

On the other hand, if any log directory fails during loading, it is possible that states have been created for some logs that directory but not others. In this case we need to call `maybeAddLogFailureEvent()` to clean up the state.